Lib/http/client.py

```diff
@@ -941,23 +941,26 @@ def _tunnel(self):
         del headers
 
         response = self.response_class(self.sock, method=self._method)
-        (version, code, message) = response._read_status()
```

Comment:
Would it be better to use `with self.response_class(...) as response:` here? I went for `.close()` because that's already used elsewhere so guaranteed to be on custom `response_class` implementations, can we rely on the context manager interface also?

Reply:
It is reasonable in `main` to use the context manager, but backports for the bugfix are better off with the `try/finally+close()` pattern _just in case_ existing applications have custom response_class implementations that don't implement the context.  So lets just leave it as is, consistent with the rest of the code.

All uses could be modernized in a separate main-only PR if ever desired.